packages/core/src/BoltTool.ts

```diff
@@ -0,0 +1,85 @@
+import path from 'path';
+import fs from 'fs-extra';
+
+import { Tool, ToolType, Package, PackageJSON, Packages, InvalidMonorepoError } from './Tool';
+import { expandPackageGlobs, expandPackageGlobsSync } from "./expandPackageGlobs";
+
+export interface BoltPackageJSON extends PackageJSON {
+    bolt?: {
+        workspaces?: string[]
+    }
+}
+
+export const BoltTool: Tool = {
+    type: 'bolt',
+
+    async isMonorepoRoot(directory: string): Promise<boolean> {
+        try {
+            const pkgJson = (await fs.readJson(path.join(directory, "package.json"))) as BoltPackageJSON;
+            if (pkgJson.bolt && pkgJson.bolt.workspaces) {
+                return true;
+            }
+        } catch (err) {
+            if (err.code !== "ENOENT") {
+                throw err;
+            }
+        }
+        return false;
+    },
+
+    isMonorepoRootSync(directory: string): boolean {
+        try {
+            const pkgJson = fs.readJsonSync(path.join(directory, "package.json")) as BoltPackageJSON;
+            if (pkgJson.bolt && pkgJson.bolt.workspaces) {
+                return true;
+            }
+        } catch (err) {
+            if (err.code !== "ENOENT") {
+                throw err;
+            }
+        }
+        return false;
+    },
+
+    async getPackages(directory: string): Promise<Packages> {
+        try {
+            const pkgJson = (await fs.readJson(path.join(directory, "package.json"))) as BoltPackageJSON;
+            const packageGlobs: string[] = pkgJson.bolt!.workspaces!;
```

Comment:
This could lead to throwing something different than `InvalidMonorepoError` and if we have such a specialized error I would expect that to be thrown in such a case.